src/components/EstimateDate.tsx

```diff
@@ -0,0 +1,135 @@
+import { Input, useClickOutside } from '@taskany/bricks';
+import { IconXSolid } from '@taskany/icons';
+import { useState, useRef, useCallback, Dispatch, SetStateAction, useEffect } from 'react';
+import InputMask from 'react-input-mask';
+
+import { useLocale } from '../hooks/useLocale';
+import { currentLocaleDate, createValue, parseLocaleDate, createLocaleDate } from '../utils/dateTime';
+import { Option, Estimate } from '../types/estimate';
+
+import { EstimateOption } from './EstimateOption';
+
+interface EstimateDateProps {
+    mask: string;
+    placeholder: string;
+    option: Omit<Option, 'clue'>;
+    value?: Estimate;
+    readOnly?: boolean;
+    onChange?: (value?: Estimate) => void;
+    setReadOnly: Dispatch<
+        SetStateAction<{
+            year: boolean;
+            quarter: boolean;
+            date: boolean;
+        }>
+    >;
+}
+
+export const EstimateDate: React.FC<EstimateDateProps> = ({
+    mask,
+    placeholder,
+    option,
+    value,
+    readOnly,
+    onChange,
+    setReadOnly,
+}) => {
+    const locale = useLocale();
+    const currentDate = currentLocaleDate({ locale });
+    const [fullDate, setFullDate] = useState(currentDate);
+    const ref = useRef(null);
+
+    useClickOutside(ref, () => {
+        if (!fullDate.includes('_')) return;
```

Comment:
Very fragile. You base condition on mask type in external library. It's better to check value. You can save it without mask.